composer/core/evaluator.py

```diff
@@ -85,7 +90,12 @@ def should_eval(state: State, event: Event):
                         last_batch_seen = state.timestamp.batch
                         return True
             elif state.max_duration.unit == TimeUnit.TOKEN and event == Event.BATCH_END:
-                raise ValueError(f'Evaluation interval of type `dur` is not supported yet for max_duration as `tok`')
+                tokens_per_interval = math.ceil(state.max_duration.value * eval_interval)
```

Comment:
can we simplify the sample logic above to also be like this?

Reply:
we _can_, but the logic is slightly different so it would be a behavior change. wdyt?

Reply:
I'd personally rather have consistent behavior